### PR TITLE
Close Bottom edge

### LINKED ISSUE #410: Editor: card search panel can never be hidden
Once the card search panel comes up, there is no way to hide it again - even when leaving the edit mode, the panel does not disappear:

bildschirmfoto 2018-11-06 um 12 37 51

The simplest way to fix that would be adding a _"close [x]"_ button to the top right or so. @dnaidionov did you have anything specific in mind?

### TODO

- [ ] add a _"close [x]"_ button to the card selector that closes the panel
- [ ] auto-close the card selector when the user leaves edit mode

### PR DESCRIPTION
**Pull request by jorgelainfiesta**
Fixes #410

### CODE REVIEW

File: packages/card-picker/package.json [hunk 1]
```diff
@@ -42,12 +42,13 @@
     "ember-cli-sri": "^2.1.1",
     "ember-cli-template-lint": "^1.0.0-beta.1",
     "ember-cli-uglify": "^2.1.0",
+    "ember-data": "^3.5.0",
```
Comment (Turbo87) [inline packages/card-picker/package.json]: why is Ember Data needed for this PR?

File: packages/card-picker/package.json [hunk 2]
```diff
@@ -42,12 +42,13 @@
...
     "ember-disable-prototype-extensions": "^1.1.3",
     "ember-export-application-global": "^2.0.0",
     "ember-load-initializers": "^1.1.0",
     "ember-maybe-import-regenerator": "^0.1.6",
     "ember-resolver": "^5.0.1",
-    "ember-source": "~3.4.0",
+    "ember-source": "~3.2.0",
```
Comment (Turbo87) [inline packages/card-picker/package.json]: I'm assuming this is because of #450, but how is that related to this PR?

File: packages/card-picker/tests/dummy/config/optional-features.json [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-  "jquery-integration": false
+  "jquery-integration": true
```
Comment (Turbo87) [inline packages/card-picker/tests/dummy/config/optional-features.json]: why do we need jQuery for this?

File: packages/card-picker/tests/acceptance/close-button-test.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import { module, test } from 'qunit';
+import { visit, click, waitFor, waitUntil, find} from '@ember/test-helpers';
+import { setupApplicationTest } from 'ember-qunit';
+
+module('Acceptance | close button', function(hooks) {
+  setupApplicationTest(hooks);
```
Comment (Turbo87) [inline packages/card-picker/tests/acceptance/close-button-test.js]: any particular reason for using an application test over a rendering/integration test?

File: packages/card-picker/addon/templates/components/card-picker-toolbox.hbs [hunk 5]
```diff
@@ -1,10 +1,20 @@
...
+  </button>
 </div>
 
 <div class="cards-list">
   {{#cardstack-search query=query as |content|}}
-    <div role="button" onclick={{action "select" content}}>
+    <div
+      role="button"
+      onclick={{action "select" content}}
+    >
```
Comment (Turbo87) [inline packages/card-picker/addon/templates/components/card-picker-toolbox.hbs]: unrelated change?

**Comment by ef4** [comment]
I'm ok with merging after @Turbo87's questions are addressed and conflicts are resolved.

File: packages/card-picker/package.json (hunk 1 above)
Comment (mansona) [reply packages/card-picker/package.json]: When you open the bottom edge it (eventually) instanciates the `cardstack-query-editor` from `@cardstack/search` which uses the `ember-data` store.

Looking at the `@cardstack/search` it seems that `ember-data` is not a dependency 🤔 I guess that should be changed.

@ef4 or @habdelra any idea why we're using a particular sha as an ember-data dependency on `@cardstack/search`? https://github.com/cardstack/cardstack/blob/master/packages/search/package.json#L44

File: packages/card-picker/package.json (hunk 2 above)
Comment (mansona) [reply packages/card-picker/package.json]: Yes this was in relation to #450, more so the `ember-elsewhere` version bump than the `ember-let` issue. We can revert once we get a release of `@cardstack/edges`

Comment (Turbo87) [reply packages/card-picker/package.json]: shouldn't that be unnecessary due to the use of `lerna`?

Comment (mansona) [reply packages/card-picker/package.json]: eh... I don't think so 🤔 just because it explicitly depends on a particular version of edges in this package.json https://github.com/cardstack/cardstack/pull/451/files#diff-d6818a352bbaab3106246009a428b3e7R25

but then again I don't know if that is true 😂

File: packages/card-picker/tests/dummy/config/optional-features.json (hunk 3 above)
Comment (mansona) [reply packages/card-picker/tests/dummy/config/optional-features.json]: This is because we need ember-data and it's still needed for now 👍

File: packages/card-picker/tests/dummy/app/router.js [hunk 6]
```diff
@@ -7,6 +7,7 @@ const Router = EmberRouter.extend({
 });
 
 Router.map(function() {
+  this.route('close-button');
```
Comment (mansona) [inline packages/card-picker/tests/dummy/app/router.js]: @jorgelainfiesta can we just get this changed to the application template please? Just so that when people spin up the dummy app for this it shows them something without needing to go to the `close-button` route

File: packages/card-picker/addon/templates/components/card-picker-toolbox.hbs (hunk 5 above)
Comment (Turbo87) [reply packages/card-picker/addon/templates/components/card-picker-toolbox.hbs]: how did it pass before this PR?

File: packages/card-picker/package.json (hunk 1 above)
Comment (habdelra) [reply packages/card-picker/package.json]: I’m unsure, @ef4 would have to answer that.

File: packages/card-picker/package.json (hunk 2 above)
Comment (jorgelainfiesta) [reply packages/card-picker/package.json]: I tried to set `ember-source` back to `3.4.0` and do `lerna bootstrap` however the bottom edge simply stops working without throwing any errors.

File: packages/card-picker/addon/templates/components/card-picker-toolbox.hbs (hunk 5 above)
Comment (jorgelainfiesta) [reply packages/card-picker/addon/templates/components/card-picker-toolbox.hbs]: I actually don't know, I guess it was before template linting was enabled?

File: packages/card-picker/tests/dummy/app/router.js (hunk 6 above)
Comment (jorgelainfiesta) [reply packages/card-picker/tests/dummy/app/router.js]: Sure, I've moved everything to the application route.

File: packages/card-picker/package.json (hunk 2 above)
Comment (Turbo87) [reply packages/card-picker/package.json]: make sure to rebase. this should have been fixed by #453
